elastic/apm-server
# Set metricset.name for breakdown metrics

## Motivation/summary

Introduce model/modelprocessor.SetMetricsetName, which sets metricset.name for well-known (breakdown) agent metrics.

- Transaction breakdown metric sets (`transaction.breakdown.count`, etc.) will be named "transaction_breakdown".
- Span breakdown metric sets (`span.self_time.*`) will be named "span_breakdown".
- All other agent metric sets will be named "app".

## Checklist


## How to test these changes

1. Instrument an app with any Elastic APM agent that supports breakdown metrics (should be all current versions).
2. Send some requests, wait >30s for the agent to send breakdown metrics.
3. Check the breakdown and system/process/runtime metric docs in Elasticsearch have a `metricset.name` field with the expected names.

## Related issues

Closes #4392
